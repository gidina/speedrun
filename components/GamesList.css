.games-title {
    text-transform: uppercase;
}

.games-list {
    list-style-type: none;
}