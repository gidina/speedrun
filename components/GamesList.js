import React from "react";
import Link from 'next/link';
import PropTypes from 'prop-types';

import "./index.css";

const GamesList = props => (
    <div>
        <h1 className="games-title">Games</h1>
        <ul className="games-list">
            {props.games.map((game) => (
                <li key={game.id} className="game">
                    <img alt="" className="game-img" src={game.assets['cover-small'].uri} />
                    <Link as={`/game/${game.id}`} href={`/game?id=${game.id}`}>
                        <a className="game-title">{game.names.international}</a>
                    </Link>
                </li>
            ))}
        </ul>
    </div>
);

export default GamesList;

GamesList.propTypes = {
    games: PropTypes.array.isRequired
};