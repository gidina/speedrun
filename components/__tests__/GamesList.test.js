import React from 'react';
import renderer from 'react-test-renderer';
import GamesList from '../GamesList';

jest.mock('next/link', () => "Link-mock");

it('Renders correctly', () => {
    const games = [
        {
            id: "game-1", 
            name: "international-name",
            logoUrl: "cover-small-uri"
        },
        {
            id: "game-2", 
            name: "international-name-2",
            logoUrl: "cover-small-uri-2"
        }
    ];
    const tree = renderer
        .create(<GamesList games={games} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});