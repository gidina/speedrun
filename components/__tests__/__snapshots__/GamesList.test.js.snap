// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  <h1
    className="games-title"
  >
    Games
  </h1>
  <ul
    className="games-list"
  >
    <li
      className="game"
    >
      <img
        alt=""
        className="game-img"
        src="cover-small-uri"
      />
      <Link-mock
        as="/game/game-1"
        href="/game?id=game-1"
      >
        <a
          className="game-title"
        >
          international-name
        </a>
      </Link-mock>
    </li>
    <li
      className="game"
    >
      <img
        alt=""
        className="game-img"
        src="cover-small-uri-2"
      />
      <Link-mock
        as="/game/game-2"
        href="/game?id=game-2"
      >
        <a
          className="game-title"
        >
          international-name-2
        </a>
      </Link-mock>
    </li>
  </ul>
</div>
`;
